tests/unit/provider/akamai/test_serviceController.py

```diff
@@ -0,0 +1,111 @@
+        self.policy_client = self.driver.policy_api_client
+        self.ccu_client = self.driver.ccu_api_client
+        print("3")
+        self.driver.provider_name = 'Akamai'
+        self.driver.http_conf_number = 1
+        self.driver.akamai_https_access_url_suffix = str(uuid.uuid1())
+        self.driver.akamai_http_access_url_suffix = str(uuid.uuid1())
+        self.san_cert_cnames = [str(x) for x in range(7)]
+        self.driver.san_cert_cnames = self.san_cert_cnames
+        self.driver.metrics_resolution = 86400
+        self.controller = services.ServiceController(self.driver)
+        service_id = str(uuid.uuid4())
+        domains_old = domain.Domain(domain='cdn.poppy.org')
+        current_origin = origin.Origin(origin='poppy.org')
+        self.service_obj = Service(service_id=service_id,
+                                   name='poppy cdn service',
+                                   domains=[domains_old],
+                                   origins=[current_origin],
+                                   flavor_id='cdn')
+
+    @ddt.file_data('data_update_service.json')
+    def test__get_provider_access_url_shared_custom(self, service_json):
+        provider_service_id = []
+        for domain_obj in service_json.get('domains', []):
+            certificates = ['custom', 'shared']
+            for certificate in certificates:
+                provider_service_id.append(
+                    {
+                        'policy_name': domain_obj['domain'],
+                        'protocol': domain_obj.get('protocol', 'https'),
+                        'certificate': domain_obj.get('certificate', certificate)
+                    }
+                )
+        dp = 'test.xxxx.secure.raxcdn.com'
+        edge_host_name = None
+        https_url = akamai_service._get_provider_access_url(domain_obj, dp, edge_host_name)
+        if domain_obj.get(certificate) == 'shared':
+            expected_out = 'xxxx.secure.raxcdn.com'
+        else:
+            expected_out = 'test.xxxx.secure.raxcdn.com'
+        self.assertEqual(https_url, expected_out)
+
+    @ddt.file_data('data_update_service.json')
+    def test__get_provider_access_url_san_sni(self, service_json):
+        provider_service_id = []
+        for domain_obj in service_json.get('domains', []):
+            certificates = ['san', 'sni']
+            for certificate in certificates:
+                provider_service_id.append(
+                    {
+                        'policy_name': domain_obj['domain'],
+                        'protocol': domain_obj.get('protocol', 'https'),
+                        'certificate': domain_obj.get('certificate', certificate)
+                    }
+                )
+
+        dp = 'test.abc.com'
+        edge_host_name = None
+        https_url = akamai_service._get_provider_access_url(domain_obj, dp, edge_host_name)
+        expected_out = 'test.abc.com.raxcdn.com.edgekey.net'
+        self.assertEqual(https_url, expected_out)
```

Comment:
This could be more readable without an additional var:  `self.assertEqual(https_url,  'test.abc.com.raxcdn.com.edgekey.net')`.